.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,99 @@
+name: Deploy & Test Deployment
+
+permissions:
+  id-token: write  # required for requesting the JWT
+  contents: read  # required for actions/checkout
+
+on:
+  # run this workflow when the `Distribute` workflow completes
+  workflow_run:
+    workflows: [Distribute]
+    types: [completed]
+  # ... or manually. 
+  workflow_dispatch:
+  
+  # remove later
+  push:
+    branches:
+      - "feat/add-integration-tests"
+    
+jobs:
+  # run a job on a successful `Workflow` run. 
+  on-successful-release:
+    runs-on: ubuntu-latest
+    env:
+      AWS_DEFAULT_REGION: ${{ secrets.AWS_DEFAULT_REGION_DEPLOY }}
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID_DEPLOY }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY_DEPLOY }}
+
+
+    steps:
+
+    - name: Checkout repository # for runs after `Distribute`, pulling from main will always give the latest eoapi-cdk. 
+      uses: actions/checkout@v3
+
+    - name: Set up python
+      uses: actions/setup-python@v2
+      with:
+        cache: pip
+
+    - name: Set up node
+      uses: actions/setup-node@v3
+      with:
+        node-version: '18'
+      
+    - name: Install deployment environment
+      working-directory: integration_tests/cdk
+      run: |
+        python -m venv .deployment_venv
+        source .deployment_venv/bin/activate
+        pip install -r requirements.txt
+        pip install --upgrade eoapi-cdk # make sure to take the latest version of eoapi-cdk since we're integration-testing the newest release. 
```

Comment:
Are we testing the latest available version on pypi or the one in this repo? 

It might be `dangerous` to test the version on pypi because there could be a `lag` from the push to the publication

Reply:
> How about setting up this workflow so that it waits (with a timeout) for the latest pypi to match for the latest github tag?

Sure, but then what will it means if the tests fails?

Reply:
Ah 🤦‍♂️ doesn't make much sense indeed. We should be doing

build -> integration test -> release

Reply:
lets see if @sharkinsspatial or @alukach have opinions

Reply:
Could be missing something from the CI logic, but one way to do it could be to add a [`job` named `integration-test` here](https://github.com/developmentseed/eoapi-cdk/blob/a1081a22e034d50f89d248d36844d598d45d5c69/.github/workflows/distribute.yaml#L11) that reuses the `package` job's output. We then add `integration-test` as a dependency to both subsequent `distribute-*` jobs so that we're sure releases happen only if integration-tests ran.

Reply:
FYI, to illustrate the idea, I pushed another commit (can always revert). See https://github.com/developmentseed/eoapi-cdk/pull/69/commits/701d85f51a5359cc31d07e37b3e004ea205b9302.

Reply:
@emileten I would recommend relying on the local python modules from the output of the TS build from the `package` output as you outlined in https://github.com/developmentseed/eoapi-cdk/commit/4ff82c23c0a09f7b0cd5d42308a3f851f9757a87.